### PR TITLE
Product -Detail

### PR DESCRIPTION
**Pull request by jaypal412**
## Motivation and Context

Product Detail UI For the Mobile  View.
## Describe your changes

Product Detail UI
- Added a similar product slider.
- Added product description.
- Added price section with add to cart button.

## Any further comments?

------

## Checklist

- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

------
Dear Gatekeeper,

Please make sure that the commits that will be merged or rebased into the base
branch are [formatted according to our standards][commit-style].

> The only additional requirement is that the PR number must appear in the
> commit title in brackets: `(#XXX)`

[commit-style]: https://github.com/aviabird/snitch/blob/develop/CONTRIBUTING.md#styling

### CODE REVIEW

File: app/core/services/product.service.ts [hunk 1]
```diff
@@ -32,26 +32,30 @@ export class ProductService {
...
-				`https://app.ofypets.com/api/v1/products?${keyword}&per_page=20&data_set=small`
-			)
-	}
+  getProduct(id: string): Observable<Product> {
+    return this.http
+      .get<Product>(
+        `api/v1/products/${id}?data_set=large`
+      )
+  }
+  getProducts(pageNumber: 1): Observable<Array<Product>> {
```
Comment (pkrawat1) [inline app/core/services/product.service.ts]: space missing between functions

File: app/home/home.component.ts [hunk 2]
```diff
@@ -15,25 +21,35 @@ export class HomeComponent implements OnInit {
   taxonImageLink;
   searchPhrase: string;
   searchBar;
+  public isAndroid: boolean;
+  public isIos: boolean;
+  productID;
   promoImg = "../assets/promo.png";
-  constructor(private myService: TaxonomyService) {
-
+  constructor(private myService: TaxonomyService, private ProductService: ProductService,
```
Comment (pkrawat1) [inline app/home/home.component.ts]: space missing before constructor

File: app/product/product.component.ts [hunk 3]
```diff
@@ -5,31 +5,56 @@ import { Product } from "../core/models/product";
...
-    this.store.dispatch(this.actions.getAllProducts(1));
-    this.store.select(showAllProducts).subscribe((productdata) => {
-      this.products = productdata;
-    });
+  relatedProducts$: Observable<Array<Product>>;
+  similarProducts$: Observable<Array<Product>>
+  product;
+  relatedProducts;
+  similarProducts;
+  quantity = 12;
```
[the remark this answers is not in the document]
Comment (jaypal412) [reply app/product/product.component.ts]: removed it

File: app/product/product.component.ts [hunk 4]
```diff
@@ -5,31 +5,56 @@ import { Product } from "../core/models/product";
...
+  constructor(private store: Store<State>, private actions: ProductActions, private ProductService: ProductService,
+    private route: ActivatedRoute, private router: Router, private searchActions: SearchActions) {
   }
 
   ngOnInit(): void {
+    const id = this.route.snapshot.queryParams.id;
+    this.ProductService.getProductDetail(id).subscribe(data => {
+      this.product = data;
+      console.log(this.product)
```
[the remark this answers is not in the document]
Comment (jaypal412) [reply app/product/product.component.ts]: Done

File: app/home/home.component.ts [hunk 5]
```diff
@@ -15,25 +21,35 @@ export class HomeComponent implements OnInit {
...
         this.texonomies = (result);
-
       });
-    this.myService.getProducts(1)
-      .subscribe((productdata) => {
-        this.products = productdata;
+  }
 
-      });
+  productDeatail(productID) {
```
[the remark this answers is not in the document]
Comment (jaypal412) [reply app/home/home.component.ts]: done

File: app/home/home.component.ts [hunk 6]
```diff
@@ -15,25 +21,35 @@ export class HomeComponent implements OnInit {
...
+    private route: ActivatedRoute, private router: Router, private store: Store<State>, private actions: ProductActions) {
   }
 
   ngOnInit() {
     this.extractData();
   }
 
   extractData() {
+    this.store.dispatch(this.actions.getAllProducts(1));
+    this.store.select(showAllProducts).subscribe((productdata) => {
```
[the remark this answers is not in the document]
Comment (jaypal412) [reply app/home/home.component.ts]: Added unsubscribe for this.

File: app/core/interceptors/interceptor.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class TokenInterceptor implements HttpInterceptor {
+  environment = "https://app.ofypets.com/"
```
[the remark this answers is not in the document]
Comment (jaypal412) [reply app/core/interceptors/interceptor.ts]: I will add environment structure in the next PR

File: app/home/home.component.ts (hunk 2 above)
Comment (jaypal412) [reply app/home/home.component.ts]: Done

**Pull request merged**